### PR TITLE
[Fluent 2 Theme for v8] spin button focus styles

### PR DESCRIPTION
**Pull request by mltejera**
Updating spin button to use Fluent 2 focused and disabled styles.

Updating other fields to use proper semantic slots.

Before:
image

After:
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5835524f6f0ed2e463e8cae0b5ae29252b6bfbc4'>5835524f6f0ed2e463e8cae0b5ae29252b6bfbc4</a></sub>
<div id="bundleSizeComment9423"/>

**Review by micahgodbolt — APPROVED** [review]
Thanks for the before and after screenshots, looks great!

File: packages/fluent2-theme/src/componentStyles/SpinButton.styles.ts [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
 
 export function getSpinButtonStyles(
   props: ISpinButtonStyleProps,
 ): IStyleFunctionOrObject<ISpinButtonStyleProps, ISpinButtonStyles> {
   const { theme, isFocused, disabled } = props;
   const { semanticColors } = theme;
-  const SpinButtonRootBorderColor = semanticColors.inputBorder;
-  const SpinButtonRootBorderColorHovered = semanticColors.inputBorderHovered;
-  const SpinButtonRootBorderColorFocused = semanticColors.inputFocusBorderAlt;
+  const SpinButtonRootBorderColorFocused = semanticColors.disabledBorder; // sorry for the broken semantics
```
Comment (brandonthomas) [inline packages/fluent2-theme/src/componentStyles/SpinButton.styles.ts:9]: nit: Might be helpful to include why we're not lining up here.

Comment (brandonthomas) [reply packages/fluent2-theme/src/componentStyles/SpinButton.styles.ts:9]: Mainly so if we need to change it's able to be maintained even if you aren't the one making the changes down the road.

**Review by brandonthomas — APPROVED** [review]

**Pull request merged**